src/test/rpc/LedgerRPC_test.cpp

```diff
@@ -279,6 +279,179 @@ class LedgerRPC_test : public beast::unit_test::suite
+        //5 - C516522DE274EB52CE69A3D22F66DD73A53E16597E06F7A86F66DF7DD4309173
+        //
+        {
+            //access via the legacy ledger field, keyword index values
+            Json::Value jvParams;
+            jvParams[jss::ledger] = "closed";
+            auto jrr = env.rpc("json", "ledger",
+                boost::lexical_cast<std::string>(jvParams))[jss::result];
+            BEAST_EXPECT(jrr.isMember(jss::ledger));
+            BEAST_EXPECT(jrr.isMember(jss::ledger_hash));
+            BEAST_EXPECT(jrr[jss::ledger][jss::ledger_index] == "5");
+
+            jvParams[jss::ledger] = "validated";
+            jrr = env.rpc("json", "ledger",
+                boost::lexical_cast<std::string>(jvParams))[jss::result];
+            BEAST_EXPECT(jrr.isMember(jss::ledger));
+            BEAST_EXPECT(jrr.isMember(jss::ledger_hash));
+            BEAST_EXPECT(jrr[jss::ledger][jss::ledger_index] == "5");
+
+            jvParams[jss::ledger] = "current";
+            jrr = env.rpc("json", "ledger",
+                boost::lexical_cast<std::string>(jvParams))[jss::result];
+            BEAST_EXPECT(jrr.isMember(jss::ledger));
+            BEAST_EXPECT(jrr[jss::ledger][jss::ledger_index] == "6");
+
+            // ask for a bad ledger keyword
+            jvParams[jss::ledger] = "invalid";
+            jrr = env.rpc ( "json", "ledger",
+                boost::lexical_cast<std::string>(jvParams)) [jss::result];
+            BEAST_EXPECT(jrr[jss::error]         == "invalidParams");
+            BEAST_EXPECT(jrr[jss::error_message] == "ledgerIndexMalformed");
+
+            // numeric index
+            jvParams[jss::ledger] = 4;
+            jrr = env.rpc("json", "ledger",
+                boost::lexical_cast<std::string>(jvParams))[jss::result];
+            BEAST_EXPECT(jrr.isMember(jss::ledger));
+            BEAST_EXPECT(jrr.isMember(jss::ledger_hash));
+            BEAST_EXPECT(jrr[jss::ledger][jss::ledger_index] == "4");
+
+            // numeric index - out of range
+            jvParams[jss::ledger] = 20;
+            jrr = env.rpc("json", "ledger",
+                boost::lexical_cast<std::string>(jvParams))[jss::result];
+            BEAST_EXPECT(jrr[jss::error]         == "lgrNotFound");
+            BEAST_EXPECT(jrr[jss::error_message] == "ledgerNotFound");
+        }
+
+        {
+            //access via the ledger_hash field
+            Json::Value jvParams;
+            jvParams[jss::ledger_hash] =
+                "7C3EEDB3124D92E49E75D81A8826A2E6"
+                "5A75FD71FC3FD6F36FEB803C5F1D812D";
+            auto jrr = env.rpc("json", "ledger",
+                boost::lexical_cast<std::string>(jvParams))[jss::result];
+            BEAST_EXPECT(jrr.isMember(jss::ledger));
+            BEAST_EXPECT(jrr.isMember(jss::ledger_hash));
+            BEAST_EXPECT(jrr[jss::ledger][jss::ledger_index] == "3");
+
+            // extra leading hex chars in hash will be ignored
```

Comment:
it calls this to decode:  https://github.com/ripple/rippled/blob/140666f97be156bdfe295c0e777190c970dfb81c/src/ripple/basics/base_uint.h#L342
..as for whether that is correct/desired bahavior, I don't know.